src/GraphQL.MicrosoftDI/ResolverBuilder.cs

```diff
@@ -0,0 +1,331 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL.Builders;
+using GraphQL.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// A builder for field resolvers with no extra service types.
+    /// </summary>
+    public class ResolverBuilder<TSourceType, TReturnType>
```

Comment:
Consider to move all methods from this "root" builder into `ScopedFieldBuilderExtensions` making them extension methods. Then you can remove `ResolverBuilder<TSourceType, TReturnType>` and `Resolve` extension method.

Reply:
It won’t work because WithType then would require TSourceType and TReturnType also. Very cumbersome.

Reply:
Well, yes. The idea breaks down on the language limitations 😞 .

Reply:
I think that was my original idea when I started this last year.  It just wouldn't work without a builder structure.  There was a lot of discussion about it in some of the issues way back.

Reply:
Builder acts just as closure for two generic types so all methods on it can be written without restrictions.